performance/theme_runner.rb

```diff
@@ -70,4 +70,57 @@ def compile_and_render(template, layout, assigns, page_template, template_file)
       content_for_layout
     end
   end
+
+  # the following 3 methods are for when we want to benchmark just the 'render'
+
+  # compile_tests is called in `initialize` to compile,
+  # render is called to benchmark just the render
+
+  def render
+    @compiled_tests.each do |test|
+      # layout could be nil
+      tmpl = test[:tmpl]
+      assigns = test[:assigns]
+      layout = test[:layout]
+
+      if layout
+        assigns['content_for_layout'] = tmpl.render!(assigns)
+        layout.render!(assigns)
+      else
+        tmpl.render!(assigns)
+      end
+    end
+  end
+
+  private
+
+  def compile_tests
+    # Dup assigns because will make some changes to them
+    assigns = Database.tables.dup
+
+    @compiled_tests = []
+    @tests.each do |liquid, layout, template_name|
```

Comment:
The following lines here look very similar to what the `run` method does. Can this be DRYed up somehow?